Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/nbclient/client.py", line 1319, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 686, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from myst_nb import glue

import numpy as np
import matplotlib.pyplot as plt


plt.rcParams['text.usetex'] = True
plt.rcParams['text.latex.preamble'] = r'\usepackage{{amsmath}}'
plt.rcParams.update({'font.size': 18})


diagdataset = np.random.multivariate_normal(np.array([0.5,-0.5]),np.array([[3.,0.],[0.,0.75]]), size=100).T
dataset     = np.random.multivariate_normal(np.array([0.5,-0.5]),np.array([[3,1.],[1.,0.75]]), size=100).T


with warnings.catch_warnings():
    warnings.simplefilter("ignore")
    fig,axis = plt.subplots(ncols=2,nrows=2)
    axis[0,0].scatter(diagdataset[0],diagdataset[1]) 
    axis[0,1].scatter(dataset[0],dataset[1]) 
    axis[0,0].axis('equal')
    axis[0,1].axis('equal')
    axis[0,0].set_xlim(-5,5)
    axis[0,1].set_xlim(-5,5)
    axis[0,0].set_ylim(-5,5)
    axis[0,1].set_ylim(-5,5)
    vals,vecs = np.linalg.eigh(np.cov(diagdataset))
    svecs = vecs[:,0]*vals[0],  vecs[:,1]*vals[1]
    #axis[0,0].quiver(0.5, -0.5, svecs[0][0],svecs[0][1],angles='xy', scale_units='xy', scale=1, color='red',linewidth=4.)
    #axis[0,0].quiver(0.5, -0.5, svecs[1][0],svecs[1][1],angles='xy', scale_units='xy', scale=1, color='red',linewidth=4.)
    axis[0,0].quiver(0.5, -0.5, 3,0,angles='xy', scale_units='xy', scale=1, color='red',linewidth=4.)
    axis[0,0].quiver(0.5, -0.5, 0,0.75,angles='xy', scale_units='xy', scale=1, color='red',linewidth=4.)
    axis[0,0].set_xticklabels([])
    axis[0,0].set_yticklabels([])
    vals,vecs = np.linalg.eigh(np.cov(dataset))
    svecs = vecs[:,0]*vals[0],  vecs[:,1]*vals[1]
    axis[0,1].quiver(0.5, -0.5, svecs[0][0],svecs[0][1],angles='xy', scale_units='xy', scale=1, color='red',linewidth=4.)
    axis[0,1].quiver(0.5, -0.5, svecs[1][0],svecs[1][1],angles='xy', scale_units='xy', scale=1, color='red',linewidth=4.)
    axis[0,1].set_xticklabels([])
    axis[0,1].set_yticklabels([])
    axis[1,0].text(0.4,0.7,r'$\begin{bmatrix} 3 & 0\\ 0 & \frac{3}{4} \end{bmatrix}$')
    axis[1,0].axis('off')
    axis[1,1].text(0.4,0.7,r'$\begin{bmatrix} 3 & 1\\ 1 & \frac{3}{4} \end{bmatrix}$')
    axis[1,1].axis('off')

    glue("plot",fig,display=False)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[2], line 16[0m
[1;32m     12[0m diagdataset [38;5;241m=[39m np[38;5;241m.[39mrandom[38;5;241m.[39mmultivariate_normal(np[38;5;241m.[39marray([[38;5;241m0.5[39m,[38;5;241m-[39m[38;5;241m0.5[39m]),np[38;5;241m.[39marray([[[38;5;241m3.[39m,[38;5;241m0.[39m],[[38;5;241m0.[39m,[38;5;241m0.75[39m]]), size[38;5;241m=[39m[38;5;241m100[39m)[38;5;241m.[39mT
[1;32m     13[0m dataset     [38;5;241m=[39m np[38;5;241m.[39mrandom[38;5;241m.[39mmultivariate_normal(np[38;5;241m.[39marray([[38;5;241m0.5[39m,[38;5;241m-[39m[38;5;241m0.5[39m]),np[38;5;241m.[39marray([[[38;5;241m3[39m,[38;5;241m1.[39m],[[38;5;241m1.[39m,[38;5;241m0.75[39m]]), size[38;5;241m=[39m[38;5;241m100[39m)[38;5;241m.[39mT
[0;32m---> 16[0m [38;5;28;01mwith[39;00m [43mwarnings[49m[38;5;241m.[39mcatch_warnings():
[1;32m     17[0m     warnings[38;5;241m.[39msimplefilter([38;5;124m"[39m[38;5;124mignore[39m[38;5;124m"[39m)
[1;32m     18[0m     fig,axis [38;5;241m=[39m plt[38;5;241m.[39msubplots(ncols[38;5;241m=[39m[38;5;241m2[39m,nrows[38;5;241m=[39m[38;5;241m2[39m)

[0;31mNameError[0m: name 'warnings' is not defined

